File: rust/src/bin/2015-02.rs
use std::fs;
use std::cmp::{min, max};

fn main() {
    let content = fs::read_to_string("input.txt")
        .expect("Can't read from file");

    let mut total_a = 0;
    let mut total_b = 0;

    for line in content.split('\n'){
        let mut arr = line.split('x').map(|x| x.parse::<i32>().unwrap());
        let (a, b, c) = (arr.next().unwrap(), arr.next().unwrap(), arr.next().unwrap());

        let s1 = a * b;
        let s2 = b * c;
        let s3 = c * a;

        total_a += 2 * (s1 + s2 + s3) + min(s1, min(s2, s3));
        total_b += 2 * (a + b + c - max(a, max(b, c))) + a * b * c;
    }
    println!("Part A: {}", total_a);
    println!("Part B: {}", total_b);
}

File: python/2015-02.py
from inputGetter import get_input

content = get_input(2015, 2).split('\n')

total_a = 0
total_b = 0

for line in content:
    a, b, c = [int(i) for i in line.split('x')]
    s1, s2, s3 = a * b, b * c, a * c
    
    total_a += 2 * (s1 + s2 + s3) + min(s1, s2, s3)
    total_b += 2 * (a + b + c - max(a, b, c)) + a * b * c

print(f'Part A: {total_a}')
print(f'Part B: {total_b}')